stdlib/ostruct.rb

```diff
@@ -60,10 +66,52 @@ def to_n
   def hash
     @table.hash
   end
+  
+  def table
+    @table
+  end
+  
+  def delete_field(name)
+    sym = name.to_sym
+    begin
+      singleton_class.__send__(:remove_method, sym, "#{sym}=")
+    rescue NameError
+    end
+    @table.delete sym
+  end
+  
+  def new_ostruct_member(name)
+    name = name.to_sym
+    unless respond_to?(name)
+      define_singleton_method(name) { @table[name] }
+      define_singleton_method("#{name}=") { |x| @table[name] = x }
+    end
+    name
+  end
 
+  `var ostruct_ids;`
+  
   def inspect
```

Comment:
This method is likely better implemented in JavaScript, you kind of inverted what should be JS and what should be Ruby.